File: back/node/app/connections/infrastructure/database/redis.ts
import redis, {RedisClientType} from 'redis';
import {REDIS_HOST} from "../../config/environment.js";


export const createRedisClient = () => {

    const redisHost: string = REDIS_HOST || 'localhost';

    console.log(`Connecting to Redis at ${redisHost}`);

    const client: RedisClientType = redis.createClient({
        url: `redis://${redisHost}:6379`
    })

    client.on('ready', () => console.log('Server connected to Redis'));
    client.on('error', err => console.log('Redis error: ', err + ` Tried to Connect to ${redisHost}`));

    return client;
};

File: back/go/app/connections/infrastructure/database/Redis.go
package database

import (
	"context"
	"fmt"
	"github.com/redis/go-redis/v9"
	"log"
	"os"
)

type Client struct {
	*redis.Client
}

func NewRedisClient() *Client {

	redisHost := os.Getenv("REDIS_HOST")
	if redisHost == "" {
		redisHost = "redis" // Default to "localhost"
	}
//	addr := fmt.Sprintf("%s:6379", redisHost)

	client := redis.NewClient(&redis.Options{
		Addr:     "redis:6379",
		Password: "",
		DB:       0,
	})

	return &Client{client}
}

func (c *Client) Connect() {
	ctx := context.Background()
	pong, err := c.Ping(ctx).Result()
	if err != nil {
		log.Fatalf("Error connecting to Redis: %v", err)
	}
	fmt.Println(pong)
}